app/src/components/billing/BillingContentTabs/PaymentMethods/AddPaymentMethodButton.tsx

```diff
@@ -0,0 +1,221 @@
+      size={{ base: "xl", md: "2xl" }}
+      closeOnOverlayClick={false}
+      closeOnEsc={false}
+      {...disclosure}
+    >
+      <ModalOverlay />
+      <ModalContent maxW="95%" h="auto">
+        <ModalHeader>
+          <HStack>
+            <Text>Add payment method</Text>
+          </HStack>
+        </ModalHeader>
+        <ModalCloseButton />
+        <ModalBody overflowY="auto" p={8}>
+          <Box w="full">
+            <VStack w="full" justifyContent={"center"}>
+              <Elements stripe={stripePromise} options={options}>
+                <SetupStripePaymentMethodForm disclosure={disclosure} clientSecret={clientSecret} />
+              </Elements>
+            </VStack>
+          </Box>
+        </ModalBody>
+      </ModalContent>
+    </Modal>
+  );
+};
+
+function SetupStripePaymentMethodForm({
+  disclosure,
+  clientSecret,
+}: {
+  disclosure: UseDisclosureReturn;
+  clientSecret: string;
+}) {
+  const stripe = useStripe();
+  const elements = useElements();
+  const selectedProject = useSelectedProject().data;
+  const utils = api.useContext();
+
+  // Code from stripe docs
+  const [handleAddPaymentDetail, addPaymentDetailLoading] = useHandledAsyncCallback(async () => {
+    const setupIntent = await addPaymentDetail();
+
+    if (setupIntent) {
+      switch (setupIntent.status) {
+        case "succeeded":
+          toast({
+            description: "Success! Your payment method has been saved.",
+            status: "success",
+            duration: 5000,
+            isClosable: true,
+          });
+          break;
+
+        case "processing":
+          toast({
+            description: "Processing payment details. Please wait a few minutes.",
+            status: "success",
+            duration: 5000,
+            isClosable: true,
+          });
+          break;
+
+        case "requires_payment_method":
+          toast({
+            description: "Failed to process payment details. Please try another payment method.",
+            status: "error",
+            duration: 5000,
+            isClosable: true,
+          });
+          break;
+      }
+      disclosure.onClose();
+    }
+
+    await utils.payments.getPaymentMethods.invalidate();
+  }, [utils, addPaymentDetail]);
+
+  async function addPaymentDetail() {
+    if (!stripe || !elements) {
+      return;
+    }
+
+    const { error } = await stripe.confirmSetup({
+      elements,
+      confirmParams: {
+        return_url: `${env.NEXT_PUBLIC_HOST ?? "app.openpipe.com"}/p/${
+          selectedProject?.slug ?? ""
+        }/billing/payment-methods`,
+      },
+      redirect: "if_required",
+    });
+
+    if (error) {
+      toast({
+        description: error.message,
+        status: "error",
+        duration: 5000,
+        isClosable: true,
+      });
+    } else {
+      const { setupIntent } = await stripe.retrieveSetupIntent(clientSecret);
+
+      return setupIntent;
+    }
+  }
+
+  return (
+    <div style={{ width: "100%" }}>
```

Comment:
To stay consistent with Chakra styling, use `Box` here instead.

Reply:
Yes, fixed!